File: 0647-palindromic-substrings/0647-palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:

        n = len(s)-1
        count = 0

        for i in range(len(s)):
            left, right = i, i
            while left >= 0 and right <= n and s[left] == s[right]:
                count += 1
                left -= 1
                right += 1

        for i in range(len(s)):
            left, right = i, i+1
            while left >= 0 and right <= n and s[left] == s[right]:
                count += 1
                left -= 1
                right += 1                

        return count

File: 0647-palindromic-substrings/0647-palindromic-substrings.java
class Solution {
    public int countSubstrings(String s) {

        int count = 0;
        int sLength = s.length();

        for(int i = 0; i < sLength; i++){
            int left = i;
            int right = i + 1;

            while(left >= 0 && right < sLength && s.charAt(left) == s.charAt(right)){
                count++;
                left--;
                right++;
            }
        }

        for(int i = 0 ; i < sLength; i++){
            int left = i;
            int right = i;

            while(left >= 0 && right < sLength && s.charAt(left) == s.charAt(right)){
                if(s.charAt(left) == s.charAt(right)){
                    count++;
                }
                left--;
                right++;
            }
        }        

        return count;        
    }
}